src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.CoreRT.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            return IntPtr.Zero;
+        }
+
+        private static void PrepareForAssemblyLoadContextRelease(IntPtr ptrNativeAssemblyLoadContext, IntPtr ptrAssemblyLoadContextStrong)
+        {
+        }
+
+        public static AssemblyLoadContext? GetLoadContext(Assembly assembly)
+        {
+            return Default;
+        }
+
+        private Assembly InternalLoadFromPath(string? assemblyPath, string? nativeImagePath)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal Assembly InternalLoad(byte[] arrAssembly, byte[] arrSymbols)
+        {
+            return ReflectionAugments.ReflectionCoreCallbacks.Load(arrAssembly, arrSymbols);
+        }
+
+        private static IntPtr InternalLoadUnmanagedDllFromPath(string unmanagedDllPath)
+        {
+            return InteropServices.NativeLibrary.Load(unmanagedDllPath);
+        }
+
+        private Assembly? GetFirstResolvedAssembly(AssemblyName assemblyName)
+        {
+            Assembly? resolvedAssembly = null;
+
+            Func<AssemblyLoadContext, AssemblyName, Assembly> assemblyResolveHandler = _resolving;
+
+            if (assemblyResolveHandler != null)
+            {
+                // Loop through the event subscribers and return the first non-null Assembly instance
+                foreach (Func<AssemblyLoadContext, AssemblyName, Assembly> handler in assemblyResolveHandler.GetInvocationList())
+                {
+                    resolvedAssembly = handler(this, assemblyName);
+                    if (resolvedAssembly != null)
+                    {
+                        return resolvedAssembly;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private Assembly? ResolveUsingEvent(AssemblyName assemblyName)
+        {
+            string? simpleName = assemblyName.Name;
+
+            // Invoke the AssemblyResolve event callbacks if wired up
+            Assembly? assembly = GetFirstResolvedAssembly(assemblyName);
+            return assembly;
+        }
+
+        internal IntPtr GetResolvedUnmanagedDll(Assembly assembly, string unmanagedDllName)
+        {
+            IntPtr resolvedDll = IntPtr.Zero;
+
+            Func<Assembly, string, IntPtr> dllResolveHandler = _resolvingUnmanagedDll;
+
+            if (dllResolveHandler != null)
+            {
+                // Loop through the event subscribers and return the first non-null native library handle
+                foreach (Func<Assembly, string, IntPtr>  handler in dllResolveHandler.GetInvocationList())
+                {
+                    resolvedDll = handler(assembly, unmanagedDllName);
+                    if (resolvedDll != IntPtr.Zero)
+                    {
+                        return resolvedDll;
+                    }
+                }
+            }
+
+            return IntPtr.Zero;
+        }
+
+        // This method is called by the VM.
```

Comment:
Delete the methods in this file that are not called from anywhere in CoreRT?

Reply:
I put them there because it gets rid of warnings about unused events on the shared side.

Felt like the lesser evil compared to another CoreRT-specific ifdeffing on the shared side.

Do you think ifdef/pragma warning on the shared side is better?

Reply:
I would create a single method in the CoreRT-specific file that references the events and has comment that it is to silence the warnings.

Reply:
Yeah, looks better.